tests/models.py

```diff
@@ -1038,6 +1038,21 @@ def test_zip(self):
         dagbag.process_file(os.path.join(TEST_DAGS_FOLDER, "test_zip.zip"))
         self.assertTrue(dagbag.get_dag("test_zip_dag"))
 
+    def test_process_file_cron_validity_check(self):
+        """
+        test if an invalid cron expression
+        as schedule interval can be identified
+        """
+        invalid_dag_files = ["test_invalid_cron.py", "test_zip_invalid_cron.zip"]
+        dagbag = models.DagBag()
+
+        for d in invalid_dag_files:
+            dagbag.process_file(os.path.join(TEST_DAGS_FOLDER, d))
+
+        for d in invalid_dag_files:
+            self.assertTrue(any([d in k and "Invalid Cron expression" in v
```

Comment:
NIT: I think we don't need `any`

Reply:
The statement inside `any()` is returning a List of Boolean values.
`Any()` helps check if there is at least one `True` element.

Reply:
Are we expecting more than one import error from importing the dedicated test DAG file?

Reply:
`dagbag.import_errors` is a Dict whose Key are file path and Value are exception details.

Here what I did is to check "File is the one I'm intending to check" AND "exception is the exception that I'm looking into". The Length of this Boolean value List is the same as the number of entries in metadata `import_error`, in which there may be other errors other than those I'm checking here (here I only try to identify Cron invalidity errors.).

Reply:
Hi @yrqls21 , great points!

- For your **1st point**, actually you're right. It's not really necessary to re-parse these two files, i.e. the code below can be moved.

```
for d in invalid_dag_files:
    dagbag.process_file(os.path.join(TEST_DAGS_FOLDER, d))
``` 
But I purposely keep them here to make it clear what I'm testing here.

- Your **2nd point** is great. I agree it makes no practical difference here, given `n` and `m` would never be too big here. But I think I will change this part as well. Always good to keep performance in mind.

Thanks for your reviewing!